From 810c0616d59809b89f5144d4afdbf70391df7a7f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alexander=20F=C3=A6r=C3=B8y?= <ahf@torproject.org>
Date: Thu, 10 Jun 2021 20:04:13 +0000
Subject: [PATCH] Fix compilation on OpenSUSE.

This patch fixes a build error with GCC 7.x which doesn't seem to accept
const int's as constants in macro initialization.

See: tpo/core/tor#40410
---
 src/feature/dirclient/dirclient.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/feature/dirclient/dirclient.c b/src/feature/dirclient/dirclient.c
index 74c6452061..a2b20621a0 100644
--- a/src/feature/dirclient/dirclient.c
+++ b/src/feature/dirclient/dirclient.c
@@ -1907,8 +1907,7 @@ dir_client_decompress_response_body(char **bodyp, size_t *bodylenp,
   /* If we're pretty sure that we have a compressed directory, and
    * we didn't manage to uncompress it, then warn and bail. */
   if (!plausible && !new_body) {
-    const int LOG_INTERVAL = 3600;
-    static ratelim_t warning_limit = RATELIM_INIT(LOG_INTERVAL);
+    static ratelim_t warning_limit = RATELIM_INIT(60 * 60);
     log_fn_ratelim(&warning_limit, LOG_WARN, LD_HTTP,
            "Unable to decompress HTTP body (tried %s%s%s, on %s).",
            description1,
-- 
GitLab

